#  Titanic DataSet Work Sheet


In [1]:
import pandas as pd  # Kütüphanemizi İmport Edelim

In [2]:
df=pd.read_csv('titanic.csv') # 'titanic.csv' Dosyasını pd (Pandas) kütüphanesi aracılığıyla oku

In [3]:
df # Dosya İçeriğini Görmek İçin Veri Setimizi Çağıralım.

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

In [4]:
df.head() # Default Halinde İlk 5 Satırı Getirdi.

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [5]:
df.info() # Veri Setimiz Hakkında Genel Bilgileri Özet Olarak Bize Verir.

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


NaN : Boş Bilgilerin Olduğu Değer.  Örneğin; Age Sütununda 714 Tane Dolu Değer Var. Bizim Toplam 891 Satırımız Olduğu İçin, Herbir Satır da Bir Yolcunun Verisini Gösterdiği İçin Toplam Boş Bilgi Sayısı 177 (891-714)

!! Boş Verileri Ortalama, Medyan vb. İstatistiksel Hesaplamalarla Gerçeğe En Yakın Değerlerle Doldurmaya Çalışacağız. Bazı Durumlarda da Boş Veri Sayısı Çok Az Olabilir (Birkaç Tane). Bu Durumda da Bu Boş Verileri Görmezden Gelebiliriz. Ya da Veri Setinde Bizden İstenen Data Type Yerine Farklı Bir Data Type Kullanılmış Olabilir (İnteger Olması Gereken Bir Yerde String İfade Kullanılması Gibi). Bu Durumda Yanlış ya da Eksik Yazılan Data Type İçin Bizden İstenen Data Type Dönüştürmemiz Gerekir. Bu Gibi İşlemlere "DATA MANIPULATION" Denir.

In [6]:
df.head(3) # İlk 3 Satırı Getirdi.

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S

In [7]:
df.tail() # Default Halinde Son 5 Satırı Getirdi.

PassengerId  Survived  Pclass                                      Name  \
886          887         0       2                     Montvila, Rev. Juozas   
887          888         1       1              Graham, Miss. Margaret Edith   
888          889         0       3  Johnston, Miss. Catherine Helen "Carrie"   
889          890         1       1                     Behr, Mr. Karl Howell   
890          891         0       3                       Dooley, Mr. Patrick   

        Sex   Age  SibSp  Parch      Ticket   Fare Cabin Embarked  
886    male  27.0      0      0      211536  13.00   NaN        S  
887  female  19.0      0      0      112053  30.00   B42        S  
888  female   NaN      1      2  W./C. 6607  23.45   NaN        S  
889    male  26.0      0      0      111369  30.00  C148        C  
890    male  32.0      0      0      370376   7.75   NaN        Q

In [8]:
df.tail(3) # Son 3 Satırı Getirdi.

PassengerId  Survived  Pclass                                      Name  \
888          889         0       3  Johnston, Miss. Catherine Helen "Carrie"   
889          890         1       1                     Behr, Mr. Karl Howell   
890          891         0       3                       Dooley, Mr. Patrick   

        Sex   Age  SibSp  Parch      Ticket   Fare Cabin Embarked  
888  female   NaN      1      2  W./C. 6607  23.45   NaN        S  
889    male  26.0      0      0      111369  30.00  C148        C  
890    male  32.0      0      0      370376   7.75   NaN        Q

In [9]:
df.sample() # Default Olarak Rasgele Bir Satır Getirir.

PassengerId  Survived  Pclass                       Name   Sex   Age  \
266          267         0       3  Panula, Mr. Ernesti Arvid  male  16.0   

     SibSp  Parch   Ticket     Fare Cabin Embarked  
266      4      1  3101295  39.6875   NaN        S

In [10]:
df.sample(3) # Rasgele Olarak 3 Satır Getirdi.

PassengerId  Survived  Pclass                          Name   Sex   Age  \
753          754         0       3            Jonkoff, Mr. Lalio  male  23.0   
244          245         0       3          Attalah, Mr. Sleiman  male  30.0   
17            18         1       2  Williams, Mr. Charles Eugene  male   NaN   

     SibSp  Parch  Ticket     Fare Cabin Embarked  
753      0      0  349204   7.8958   NaN        S  
244      0      0    2694   7.2250   NaN        C  
17       0      0  244373  13.0000   NaN        S

In [11]:
df.columns # Data Setinin İçeriğine Göre Bazen Onlarca Sütun Olabilir. Bu Şekilde Tüm Sütunları Görebiliriz.

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

In [12]:
df.shape # Data Setinin Kaç Satır Kaç Sütundan Oluştuğunu Görmemizi Sağlar. (891 Satır, 12 Sütun)

(891, 12)

In [13]:
df.ndim # Data Setinin Kaç Boyutlu Olduğunu Gösterir. Satır ve Sütunlardan Oluşan Data Seti 2 Boyutlu Olarak Tanımlanır.

2

In [14]:
df.describe() # Veri Setimiz Hakkında İstatiksel Bilgileri Verir.

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

In [15]:
df.isnull() # Eksik Bilgileri ( NaN ) Gösterir.

PassengerId  Survived  Pclass   Name    Sex    Age  SibSp  Parch  Ticket  \
0          False     False   False  False  False  False  False  False   False   
1          False     False   False  False  False  False  False  False   False   
2          False     False   False  False  False  False  False  False   False   
3          False     False   False  False  False  False  False  False   False   
4          False     False   False  False  False  False  False  False   False   
..           ...       ...     ...    ...    ...    ...    ...    ...     ...   
886        False     False   False  False  False  False  False  False   False   
887        False     False   False  False  False  False  False  False   False   
888        False     False   False  False  False   True  False  False   False   
889        False     False   False  False  False  False  False  False   False   
890        False     False   False  False  False  False  False  False   False   

      Fare  Cabin  Embarked  
0    False   True     False  
1    False  False     False  
2    False   True     False  
3    False  False     False  
4    False   True     False  
..     ...    ...       ...  
886  False   True     False  
887  False  False     False  
888  False   True     False  
889  False  False     False  
890  False   True     False  

[891 rows x 12 columns]

In [17]:
df.isnull().sum() # Sütun Sütun Kaç Tane NaN Değer Olduğunu Gösterir.

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

In [4]:
df.Survived # Survived Sütunu'nu Çağırdık. 0 (False) = Hayatta Değil, 1 (True) = Hayatta


0      0
1      1
2      1
3      1
4      0
      ..
886    0
887    1
888    0
889    1
890    0
Name: Survived, Length: 891, dtype: int64

In [21]:
df.Survived.value_counts() # Toplam Kaç Kişi Hayatta, Kaç Kişi Ölmüş Bunun Bilgilerini Verir.

Survived
0    549
1    342
Name: count, dtype: int64

In [22]:
df.Survived.value_counts(normalize=True) 
# Default Değeri 'False', Toplam Oran Olarak Bilgi Verir. %61 Hayatta Değilken %39 Hayatta Kalmış.

Survived
0    0.616162
1    0.383838
Name: proportion, dtype: float64

In [26]:
df.Embarked.unique()
# Toplam Kaç Farklı Liman Olduğunu Bulmamızı Sağlar.
# ‘unique()’ verilen verinin kaç adet ‘eşşiz’ verisi olduğunu bize verir.

array(['S', 'C', 'Q', nan], dtype=object)

In [35]:
df['Survived'] == 1 # Survived Sütunu İçerisnde Bulunan Her İndeksi True / False Şeklinde Getirir. 

0      False
1       True
2       True
3       True
4      False
       ...  
886    False
887     True
888    False
889     True
890    False
Name: Survived, Length: 891, dtype: bool

In [36]:
df[df['Survived'] == 1]

PassengerId  Survived  Pclass  \
1              2         1       1   
2              3         1       3   
3              4         1       1   
8              9         1       3   
9             10         1       2   
..           ...       ...     ...   
875          876         1       3   
879          880         1       1   
880          881         1       2   
887          888         1       1   
889          890         1       1   

                                                  Name     Sex   Age  SibSp  \
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
8    Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                  Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   
..                                                 ...     ...   ...    ...   
875                   Najib, Miss. Adele Kiamie "Jane"  female  15.0      0   
879      Potter, Mrs. Thomas Jr (Lily Alexenia Wilson)  female  56.0      0   
880       Shelley, Mrs. William (Imanita Parrish Hall)  female  25.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
889                              Behr, Mr. Karl Howell    male  26.0      0   

     Parch            Ticket     Fare Cabin Embarked  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
8        2            347742  11.1333   NaN        S  
9        0            237736  30.0708   NaN        C  
..     ...               ...      ...   ...      ...  
875      0              2667   7.2250   NaN        C  
879      1             11767  83.1583   C50        C  
880      1            230433  26.0000   NaN        S  
887      0            112053  30.0000   B42        S  
889      0            111369  30.0000  C148        C  

[342 rows x 12 columns]

In [32]:
df1= df[df['Survived'] == 1] 

df1

# Sadece Hayatta Kalanları Çağırdığımız Bir Veri Seti Oluşturduk ve bir sonraki örnekte kullanmak için df1 olarak yeni atama yaptık.


PassengerId  Survived  Pclass  \
1              2         1       1   
2              3         1       3   
3              4         1       1   
8              9         1       3   
9             10         1       2   
..           ...       ...     ...   
875          876         1       3   
879          880         1       1   
880          881         1       2   
887          888         1       1   
889          890         1       1   

                                                  Name     Sex   Age  SibSp  \
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
8    Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                  Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   
..                                                 ...     ...   ...    ...   
875                   Najib, Miss. Adele Kiamie "Jane"  female  15.0      0   
879      Potter, Mrs. Thomas Jr (Lily Alexenia Wilson)  female  56.0      0   
880       Shelley, Mrs. William (Imanita Parrish Hall)  female  25.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
889                              Behr, Mr. Karl Howell    male  26.0      0   

     Parch            Ticket     Fare Cabin Embarked  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
8        2            347742  11.1333   NaN        S  
9        0            237736  30.0708   NaN        C  
..     ...               ...      ...   ...      ...  
875      0              2667   7.2250   NaN        C  
879      1             11767  83.1583   C50        C  
880      1            230433  26.0000   NaN        S  
887      0            112053  30.0000   B42        S  
889      0            111369  30.0000  C148        C  

[342 rows x 12 columns]

In [33]:
df[df['Age'] == 29.0] # Sadece Yaşı 29 Olan Yolcuları Çağırdık.

PassengerId  Survived  Pclass  \
53            54         1       2   
66            67         1       2   
81            82         1       3   
90            91         0       3   
117          118         0       2   
133          134         1       2   
231          232         0       3   
242          243         0       2   
251          252         0       3   
255          256         1       3   
336          337         0       1   
361          362         0       2   
422          423         0       3   
455          456         1       3   
477          478         0       3   
510          511         1       3   
567          568         0       3   
713          714         0       3   
730          731         1       1   
782          783         0       1   

                                                  Name     Sex   Age  SibSp  \
53   Faunthorpe, Mrs. Lizzie (Elizabeth Anne Wilkin...  female  29.0      1   
66                        Nye, Mrs. (Elizabeth Ramell)  female  29.0      0   
81                         Sheerlinck, Mr. Jan Baptist    male  29.0      0   
90                                Christmann, Mr. Emil    male  29.0      0   
117                    Turpin, Mr. William John Robert    male  29.0      1   
133      Weisz, Mrs. Leopold (Mathilde Francoise Pede)  female  29.0      1   
231                           Larsson, Mr. Bengt Edvin    male  29.0      0   
242                    Coleridge, Mr. Reginald Charles    male  29.0      0   
251         Strom, Mrs. Wilhelm (Elna Matilda Persson)  female  29.0      1   
255            Touma, Mrs. Darwis (Hanne Youssef Razi)  female  29.0      0   
336                          Pears, Mr. Thomas Clinton    male  29.0      1   
361                          del Carlo, Mr. Sebastiano    male  29.0      1   
422                                 Zimmerman, Mr. Leo    male  29.0      0   
455                                 Jalsevac, Mr. Ivan    male  29.0      0   
477                          Braund, Mr. Lewis Richard    male  29.0      1   
510                           Daly, Mr. Eugene Patrick    male  29.0      0   
567        Palsson, Mrs. Nils (Alma Cornelia Berglund)  female  29.0      0   
713                         Larsson, Mr. August Viktor    male  29.0      0   
730                      Allen, Miss. Elisabeth Walton  female  29.0      0   
782                             Long, Mr. Milton Clyde    male  29.0      0   

     Parch         Ticket      Fare Cabin Embarked  
53       0           2926   26.0000   NaN        S  
66       0     C.A. 29395   10.5000   F33        S  
81       0         345779    9.5000   NaN        S  
90       0         343276    8.0500   NaN        S  
117      0          11668   21.0000   NaN        S  
133      0         228414   26.0000   NaN        S  
231      0         347067    7.7750   NaN        S  
242      0    W./C. 14263   10.5000   NaN        S  
251      1         347054   10.4625    G6        S  
255      2           2650   15.2458   NaN        C  
336      0         113776   66.6000    C2        S  
361      0  SC/PARIS 2167   27.7208   NaN        C  
422      0         315082    7.8750   NaN        S  
455      0         349240    7.8958   NaN        C  
477      0           3460    7.0458   NaN        S  
510      0         382651    7.7500   NaN        Q  
567      4         349909   21.0750   NaN        S  
713      0           7545    9.4833   NaN        S  
730      0          24160  211.3375    B5        S  
782      0         113501   30.0000    D6        S

In [34]:
df1[df1['Age']==29] # Sadece Hayatta Olup da Yaşı 29 Olanları Çağırdık.

PassengerId  Survived  Pclass  \
53            54         1       2   
66            67         1       2   
81            82         1       3   
133          134         1       2   
255          256         1       3   
455          456         1       3   
510          511         1       3   
730          731         1       1   

                                                  Name     Sex   Age  SibSp  \
53   Faunthorpe, Mrs. Lizzie (Elizabeth Anne Wilkin...  female  29.0      1   
66                        Nye, Mrs. (Elizabeth Ramell)  female  29.0      0   
81                         Sheerlinck, Mr. Jan Baptist    male  29.0      0   
133      Weisz, Mrs. Leopold (Mathilde Francoise Pede)  female  29.0      1   
255            Touma, Mrs. Darwis (Hanne Youssef Razi)  female  29.0      0   
455                                 Jalsevac, Mr. Ivan    male  29.0      0   
510                           Daly, Mr. Eugene Patrick    male  29.0      0   
730                      Allen, Miss. Elisabeth Walton  female  29.0      0   

     Parch      Ticket      Fare Cabin Embarked  
53       0        2926   26.0000   NaN        S  
66       0  C.A. 29395   10.5000   F33        S  
81       0      345779    9.5000   NaN        S  
133      0      228414   26.0000   NaN        S  
255      2        2650   15.2458   NaN        C  
455      0      349240    7.8958   NaN        C  
510      0      382651    7.7500   NaN        Q  
730      0       24160  211.3375    B5        S

In [37]:
df.groupby(['Pclass', 'Sex'])
# Pclass ve Cinsiyeti Gruplandırdık Ama Neye Göre, Nasıl Bir Gruplandırma Yapacağımızı Söylemediğimiz İçin Sonuç Döndürmedi.

In [39]:
df.groupby(['Pclass', 'Sex'])['Survived'].mean()

# Bilet Sınıflarına Göre Cinsiyetleri Gruplandırdı ve Hayatta Kalanların Ortalamasını Aldı.

# Örn: 1. Sınıfta Yer Alan Kadınların Hayatta Kalma Oranı %97, Erkeklerin Hayatta Kalma Oranı %37

Pclass  Sex   
1       female    0.968085
        male      0.368852
2       female    0.921053
        male      0.157407
3       female    0.500000
        male      0.135447
Name: Survived, dtype: float64

In [41]:
df.groupby(['Pclass', 'Sex'])['Age'].median()

# Medyan en çok tekrar eden değer demektir.

# Daha Yaşlılar Daha İyi Sınıfı Tercih Ediyor Sonucunu Çıkarabiliriz.


Pclass  Sex   
1       female    35.0
        male      40.0
2       female    28.0
        male      30.0
3       female    21.5
        male      25.0
Name: Age, dtype: float64

In [42]:
df.groupby(['Pclass', 'Sex'])['Age'].mean()

# Yaş Ortalaması Arttıkça Daha İyi Sınıfın Tercih Edilmesi de Artıyor.


Pclass  Sex   
1       female    34.611765
        male      41.281386
2       female    28.722973
        male      30.740707
3       female    21.750000
        male      26.507589
Name: Age, dtype: float64